src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(
+    name,
+    health = 100,
+    hidden = false,
+  ) {
+    super(name, health);
+
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(
+    name,
+    health = 100,
+  ) {
+    super(name, health);
+  }
```

Comment:
Useless constructor.
It does the same as Annimal constructor.
you can delete it.